packages/@sanity/form-builder/src/sanity/DefaultAssetSource/DeleteAssetErrorDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import {LinkIcon} from '@sanity/icons'
+import {Box, Text, Dialog, Grid, Card} from '@sanity/ui'
+import styled from 'styled-components'
+import Preview from '../../Preview'
+import {AssetRecord} from '../../inputs/files/ImageInput/types'
+import {schema, WithReferringDocuments} from '../../legacyParts'
+
+import {SpinnerWithText} from '../../components/SpinnerWithText'
+import {IntentLink} from '../../transitional/IntentLink'
+
+interface Props {
+  asset: AssetRecord
+  onClose: () => void
+  error: Error
+}
+
+const DocumentLink = styled(IntentLink)`
+  color: inherit;
+  text-decoration: inherit;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`
+
+export function DeleteAssetErrorDialog({asset, onClose, error}: Props) {
+  return (
+    <Dialog id="asset-dialog" header="Image could not be deleted" width={2} onClose={onClose}>
+      <Card padding={4} tone="critical">
+        <Grid gap={2} style={{gridTemplateColumns: 'max-content 1fr'}}>
+          <Box padding={2}>
+            <img
+              src={`${asset.url}?w=200`}
+              style={{maxWidth: '200px'}}
+              alt="The image that is in use by the listed documents"
+            />
+          </Box>
+          <Box padding={2}>
+            <WithReferringDocuments id={asset._id}>
+              {({isLoading, referringDocuments}) => {
+                const drafts = referringDocuments.reduce(
+                  (acc, doc) =>
+                    doc._id.startsWith('drafts.') ? acc.concat(doc._id.slice(7)) : acc,
+                  []
+                )
+
+                const filteredDocuments = referringDocuments.filter(
+                  (doc) => !drafts.includes(doc._id)
+                )
+
+                if (isLoading) {
+                  return <SpinnerWithText text="Loading..." />
+                }
+                if (filteredDocuments.length === 0) {
+                  return <Text>Could not delete image: {error.message}</Text>
+                }
+                return (
+                  <>
+                    <h4 style={{margin: '0 0 1em 0'}}>
+                      {filteredDocuments.length > 1 ? (
+                        <>{filteredDocuments.length} documents are using this image</>
+                      ) : (
+                        <>One document is using this image</>
+                      )}
+                    </h4>
+                    <Box as="p" marginBottom={2}>
+                      <Text>
```

Comment:
In this case, it’s better to do `<Box marginBottom={2}><Text as="p">` – otherwise the markup will be `<p><div>` which is not completely semantic.

Reply:
Useful to know. Thanks!